### ISSUE TITLE
Where should we put submodules and submodels

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#199]
Currently we have a directory structure that looks like this:
```
Repository
    |____/drivers
    |____/samples
    |____/tools
    |____/vic_run
```
A few projects that are currently underway will be linking / coupling other models to VIC.  The question is **where do we want to put these models?**  

While we're thinking about this, I recommend we take think about where we'll put some other 

I'll propose this structure, which is widely used in Python and C open source projects:
```
Repository
    |____/docs
    |____/samples
    |____/submodels
    |    |____/submodel_a
    |    |____/submodel_b
    |____/tests
    |____/tools
    |____/vic
    |    |____/drivers
    |    |____/vic_run
```
We could also put the `/submodules` directory inside `/vic` directory.

Thoughts?

**Comment by bartnijssen** [comment]
@jhamman: Do these submodels (I'd prefer a different name, but can't come up with anything right now) share code with VIC or not?

**Comment by jhamman** [comment]
@bartnijssen:
- I don't care for the name `submodels` much either.  I'm keen on changing it.
- Presumably there would be a few functions from each submodel that VIC would use but that is probably it.  I think CN is a good example.  We certainly don't want to drop the full CN code into VIC run but we need it in the repo and a few functions from CN will be called by VIC.

As I've thought about it more, I'm revising my proposed structure to this, with the submodels under `/vic`:
```
Repository
    |____/docs
    |____/samples
    |____/tests
    |____/tools
    |____/vic
    |    |____/drivers
    |    |____/submodels
    |    |    |____/submodel_a
    |    |    |____/submodel_b
    |    |____/vic_run

**Comment by bartnijssen** [comment]
What about "addons"

**Comment by jhamman** [comment]
"addons" could work.  Other ideas:
- "plugins"
- "extensions"

**Comment by bartnijssen** [comment]
extensions is nice - I'd go with that (plug-in may be promising too much).

Bart

**Comment by jhamman** [comment]
Does anyone have any problems with making this directory rearrangement change now.  We're about to head into an active phase of coding and I think it would be nice to get this done ahead of time?

cc: @bartnijssen @tbohn

**Comment by bartnijssen** [comment]
Please go ahead

**Comment by jhamman** [comment]
closed via #228.
